### PR TITLE
Removing v1.2 from docs

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Removing v1.2 from the docs:
- removes version from dropdown
- removes LastVersion='1.2' blocks
- removes various references to v1.2 that no longer need the context

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

File: website/docs/sql-reference/date-functions/sql-dateadd.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ Switching back and forth between those SQL syntaxes usually requires a quick sca
 
 But couldn’t we be doing something better with those keystrokes, like typing out and then deleting a tweet?
 
-dbt v1.2 helps us smooth out these wrinkles of writing [SQL across data warehouses](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros).
+dbt helps us smooth out these wrinkles of writing [SQL across data warehouses](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros).
```
Comment (matthewshaver) [inline website/docs/sql-reference/date-functions/sql-dateadd.md]:
```suggestion
dbt helps smooth out these wrinkles of writing [SQL across data warehouses](https://docs.getdbt.com/reference/dbt-jinja-functions/cross-database-macros).
```

**Review by nghi-ly — COMMENTED** [review]
looks great! thanks for doing all this work! 

curious, what about these?

- should the 1.2 upgrade guide https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.2 be moved under "Older versions" ?
- should the Python compat table https://docs.getdbt.com/faqs/Core/install-python-compatibility#python-compatibility-matrix be updated so 1.2 isn't listed? 
- should the About hooks section https://docs.getdbt.com/docs/build/hooks-operations#about-hooks be updated so it doesn't mention 1.2 and is more generic (like your other updates)? there are 2 instances in that section.
- should pages like https://docs.getdbt.com/docs/core/connect-data-platform/dremio-setup be updated so all the `min_core_version` is set to the lowest supported version of Core (which is 1.6 right now)?

**Comment by matthewshaver** [comment]
@nghi-ly Great questions!

- "Older versions" only refers to deprecated versions < v1.0. While 1.0-1.5 are end of life, we haven't technically deprecated them. I think we'll leave it as is for this PR, but this is something I'll bring up to Jerco in an upcoming version meeting to get his thoughts
- For any tables or requirements we're leaving the version numbers in if they're > v1.0 since they're only EOL and not deprecated.
- Good catch! I'll fix this
- Same as above, if it's listing a requirement that other similar features would have, we'll leave it for now.

**Review by nghi-ly — APPROVED** [review]
:shipit:

**Pull request merged**